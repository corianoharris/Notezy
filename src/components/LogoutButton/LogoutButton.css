.logout-button-container {
  width: 100%;
  margin-bottom: 20px;
  text-align: right;
}



@media screen and (min-width: 1024px) {
  .logout-button-container {
    width: auto;
    margin-bottom: none;
  }
}