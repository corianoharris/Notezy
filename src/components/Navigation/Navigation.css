.navbar {
  padding-bottom: 30px;
  width: 100vw;
  height: 200px;
  background-color: #3c005a;
}

.navbar-nav {
  align-items: center;
  gap: 10px;
}

.navbar-brand {
  font-size: 1rem;
}

.nav-link, .nav-link:visited, .nav-link:link {
  padding-bottom: 5px;
  width: auto;
  color: darkgrey;
}

.nav-link:hover {
  font-weight: 700;
  padding-bottom: 5px;
  color: white;
  border-bottom: 1px solid white;
}

.nav-link:active  {
  color: white;
  font-weight: 700;
}

.active {
  font-weight: bolder;
  color: grey;
}


.greeting-container {
  position: absolute;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  top: 10px;
  right: 10px;
  width: auto;
}

.brand-link {
  display: flex;
  align-items: center;
  gap: 5px;
}

.brand-name {
  font-size: 1.725rem;
  pointer-events: none;
  cursor: default;
}

.branding-image {
  margin-left: 10px;
  width: 16px;
  height: 16px;
}

@media screen and (min-width: 480px) {
  .navbar-brand {
    font-size: 1.5rem;
  }
}

@media screen and (min-width: 990px) {
  .branding-image {
    height: 32px;
    width: 32px;
  }
}

@media screen and (max-width: 900px) {
  .navbar-nav {
    display: block;
  }
}