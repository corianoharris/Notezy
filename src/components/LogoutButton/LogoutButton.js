import React from 'react';
import { useAuth0 } from '@auth0/auth0-react';
import "./LogoutButton.css";
import Loader from "../Loader";

const LogoutButton = () => {
	const { logout, isLoading } = useAuth0();
	if (isLoading) {
		return <Loader />;
	}

	return (
		<div className='logout-button-container'>
		<button
			className="btn btn-outline-light"
			onClick={() => logout({ returnTo: window.location.origin })}>
			Log Out
			</button>
			</div>
	);
};

export default LogoutButton;