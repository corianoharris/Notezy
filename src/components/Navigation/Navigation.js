import React from 'react';
import LogoutButton from '../LogoutButton';
import './Navigation.css';

const  Navigation = () => {
	return (
		<nav className="navbar navbar-dark bg-dark navbar-expand-lg border-bottom-3 shadow ">
			<a
				href={'/'}
				className="navbar-brand links"
				style={{
					fontFamily: 'Permanent Marker',
					fontSize: '48px',
					color: 'white'
				}}>
				Notezy
			</a>
			<div className="collpase navbar-collapse">
				<ul className="navbar-nav mr-auto">
					<li className="navbar-item">
						<a
							href={'/note'}
							className="nav-link active links"
							style={{
								color: '#d3d3d3d3',
							}}>
							Create a note
						</a>
					</li>
					<li className="navbar-item">
						<a
							href={'/notes'}
							className="nav-link links"
							style={{
								color: '#d3d3d3d3',
							}}>
							Your Notes
						</a>
					</li>
				</ul>
			</div>
			<LogoutButton />
		</nav>
	);
};

export default Navigation;
