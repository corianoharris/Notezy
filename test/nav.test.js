import React from 'react';
import  Navigation from '../src/components/ Navigation';
import LogoutButton from '../src/components/LogoutButton';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

// react test renderer to test ui
it('renders correctly', () => {
	const nav = renderer.create(< Navigation />).toJSON();
	const logout = renderer.create(<LogoutButton />).toJSON();
	expect(nav).toMatchSnapshot();
	expect(logout).toMatchSnapshot();
});

// testing-library/react ti test nav links
test('renders collectins link', () => {
	const { getByText } = render(< Navigation />);
	const linkElement = getByText(
		/Create a note/i,
		/Your notes/i
	);
	expect(linkElement).toBeInTheDocument();
});
