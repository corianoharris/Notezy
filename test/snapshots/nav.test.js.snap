// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="navbar navbar-dark bg-dark navbar-expand-lg"
>
  <a
    className="navbar-brand"
    href="/"
  >
    Notezy
  </a>
  <div
    className="collpase navbar-collapse"
  >
    <ul
      className="navbar-nav mr-auto"
    >
      <li
        className="navbar-item"
      >
        <a
          className="nav-link"
          href="/note"
        >
          Note
        </a>
      </li>
      <li
        className="navbar-item"
      >
        <a
          className="nav-link"
          href="/notes"
        >
          Collections
        </a>
      </li>
    </ul>
  </div>
  <div
    className="container mt-5 mx-auto Loader text-center"
  >
    <svg
      fill="currentColor"
      height={70}
      size={70}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width={70}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
      />
    </svg>
  </div>
</nav>
`;
